### DISCUSSION TITLE
Rules for Scala 3 optional brace colons?

### DISCUSSION THREAD

**Discussion by molarmanful** [scalameta/scalafmt#3671] [Q&A]
Hello all,

Is there a rule for automatically converting e.g.

```scala
// before
List(1, 2, 3, 4).foldLeft(0)((x, y) => x + y)

// after
List(1, 2, 3, 4).foldLeft(0): (x, y) =>
  x + y
```

Perhaps taking into account `maxColumn`, or some other preferences:

```scala
// lambda
List(1, 2, 3, 4).foldLeft(0):
  (x, y) => x + y

// auto-format cases
(x: List[Option[Int]]).map:
  case Some(x) => ???
  case None => ???
```

Thanks in advance!

**Comment by tgodzik** [comment]
I don't think that currently implemented, would require a feature request or a contribution.

**Reply by molarmanful** [reply]
Sure thing, I'll put it in as a feature request. Should I file it under issues or discussions?

**Comment by kitbellew** [comment]
@molarmanful what should be done with
```
List(1, 2, 3, 4).map(x => x * x).foldLeft(0)((x, y) => x + y)
```

**Reply by molarmanful** [reply]
@kitbellew I think that would probably depend on how the formatter breaks it down, e.g. if the formatter would normally do

```scala
List(1, 2, 3, 4)
  .map(x =>
    x * x
  )
  .foldLeft(0)((x, y) =>
    x + y
  )
```

the parentheses would be removed based on that trailing/hanging parenthesis.

```scala
List(1, 2, 3, 4)
  .map: x =>
    x * x
  .foldLeft(0): (x, y) =>
    x + y
```

(this is how I've basically been doing it by hand)
